### ISSUE TITLE
How much vram needed to finetune 3b model? Is 12gb enough?

### ISSUE THREAD

**Issue by universewill** [CStanKonrad/long_llama#15]
How much vram needed to finetune 3b model? Is 12gb enough?

**Comment by CStanKonrad** [comment]
Unfortunately, 12GB is not enough to finetune the 3B model in the standard way (tuning all parameters). That is because  of optimizer variables and values for gradient accumulation. This [Hugging Face blog post](https://huggingface.co/docs/transformers/v4.20.1/en/perf_train_gpu_one#anatomy-of-models-memory) briefly describes how much each of those parts contributes to VRAM usage.
For our model, we have used a single A100 80GB GPU and usage metrics show that > 70GB of the GPU memory was allocated.

**Comment by rosnikv** [comment]
@CStanKonrad  Is it possible to fine tune 3b model using peft?
